include/revng/Pipeline/CLOption.h

```diff
@@ -6,31 +6,43 @@
 
 #include "llvm/ADT/StringMap.h"
 #include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Casting.h"
 #include "llvm/Support/CommandLine.h"
 
+#include "revng/ADT/Concepts.h"
 #include "revng/Support/Assert.h"
 
 namespace pipeline {
 
+template<typename T>
+concept PipelineOptionType = anyOf<T, bool, int, uint64_t, std::string>();
```

Comment:
I'd put as a first line of the commit message something like "Analyses options: restrict to bool, int and strings"